src/Aperture/elliptical.jl

```diff
@@ -94,3 +94,57 @@ function mask(e::EllipticalAperture; method = :center)
     ny, nx = size(e)
     return elliptical_overlap(bounds..., nx, ny, e.a, e.b, e.theta, method = method)
 end
+
+
+#######################################################
+
+"""
+    EllipticalAnnulus(x, y, a, b, θ, factor)
+    EllipticalAnnulus([x, y], a, b, θ, factor)
+
+An elliptical annulus with semi-major axis `a`, semi-minor axis `b`, angle `θ` of inner ellipse ,scaling factor `factor` to generate the outer ellipse from inner ellipse.
+`a` and `b` must be ≥ 0, factor ≥ 1, `θ` is measured in degrees counter-clockwise the standard x-axis.
+
+# Examples
+```jldoctest
+julia> ap = EllipticalAnnulus(0, 0, 10, 5, 45, 2)
+EllipticalAnnulus(0, 0, a=10, b=5, θ=25°, factor=2)
+
+```
+"""
+struct EllipticalAnnulus{T <: Number} <: AbstractAperture
+    x::T
+    y::T
+    a::T
+    b::T
+    theta::T
+    factor::T
+
+    function EllipticalAnnulus(x::T, y::T, a::T, b::T, theta::T, factor::T) where T <: Number
+        a < 0 && error("Invalid axis a=$a. a must be greater than or equal to 0")
+        b < 0 && error("Invalid axis b=$b. a must be greater than or equal to 0")
+        factor < 1 && error("Invalid scaling factor=$b. factor must be greater than or equal to 1")
+        new{T}(x, y, a, b, mod(theta, 360), factor)
+    end
+end
+
+EllipticalAnnulus(x, y, a, b, theta, factor) = EllipticalAnnulus(promote(x, y, a, b, theta, factor)...)
+EllipticalAnnulus(center::AbstractVector, a, b, theta, factor) = EllipticalAnnulus(center..., a, b, theta, factor)
+
+function Base.show(io::IO, e::EllipticalAnnulus)
+    print(io, "EllipticalAnnulus($(e.x), $(e.y), a=$(e.a), b=$(e.b), θ=$(e.theta)°, factor=$(e.factor))")
+end
+
+function bbox(e::EllipticalAnnulus)
+    xmin, xmax, ymin, ymax = bbox(EllipticalAperture(e.x, e.y, e.a * e.factor, e.b * e.factor, e.theta))
```

Comment:
I would like to see this without having to spend the allocation on creating another type. The actual code might be repetitive but it'll be more memory-efficient (and marginally faster) without the allocate.